dandischema/datacite/__init__.py

```diff
@@ -67,16 +74,23 @@
 
 def to_datacite(
     meta: Union[dict, PublishedDandiset],
+    event: str = None,
     validate: bool = False,
-    publish: bool = False,
```

Comment:
Let's ease transition via a deprecation cycle

- leave `publish` in (so code in dandi-archive could still use it as it does ATM)
  - default it to `None`. If `bool` type provided -- issue DeprecationWarning informing to use `event`, treat bool accordingly to set value of `event` while first checking if not set in parallel -- then `ValueError` -- only one should be provided
- add `event` **after** publish, so anything using it as positional arg, would not collide (we might want to add `*`)

Reply:
Actually I got a little ahead of the other PR here, this change isn't used as it is currently.

This change does open the door to "hide" findable DOIs (converting to "Registered"), so depending on how the design doc changes this might just get removed.